core/arch/arm/mm/core_mmu.c

```diff
@@ -1465,6 +1465,9 @@ bool core_mmu_add_mapping(enum teecore_memtypes type, paddr_t addr, size_t len)
 	granule = 1 << tbl_info.shift;
 	p = ROUNDDOWN(addr, granule);
 	l = ROUNDUP(len + addr - p, granule);
+
+	if (map->size < l)
+		return false;
```

Comment:
proposed comment `/* Ban overflowing virtual addresses */`

minor: can you add an empty line below? It was already missing.